lib/mosaic-objects/src/main/java/org/eclipse/mosaic/lib/objects/v2x/etsi/DenmContent.java

```diff
@@ -184,8 +189,9 @@ public String getExtendedContainer() {
     @Override
     public void toDataOutput(DataOutput dataOutput) throws IOException {
         dataOutput.writeLong(time);
-        SerializationUtils.encodeGeoPoint(dataOutput, senderPosition);
-
+        if (senderPosition != null) {
```

Comment:
This prevents the NullPointerException, yes. But the code which does the desarialization to create an object from DataInput (see line 106) no tries to read a GeoPoint but can never know if it was written or not written by this method. You can do it exactly like it is done with the `eventLocation` field.

Reply:
My IDE tells me that the constructor `public DenmContent(DataInput din) throws IOException { ... }` is never used. When I run the ReleaseBarnimIT and set breakpoints to line 106 and around, it never reaches that bit. I will ofc implement it as you said, but are you sure it is used?

Reply:
It might be used by other scenarios. The use-cases when this is actually used are rather limited, but we want to provide the possibility to serialize and deserialize this type of object.

The constructor call is hidden in `org.eclipse.mosaic.lib.objects.v2x.EncodedPayload#decodePayload()`. Here reflection is used call the constructor using the DataInput from a previously serialized object. This is something the IDE cannot know.

Reply:
Thank you for the detailed explanation - very helpful :)